bin/v-add-firewall-chain

```diff
@@ -47,8 +48,8 @@ fi
 #                    Verifications                         #
 #----------------------------------------------------------#
 
-check_args '1' "$#" 'CHAIN [PORT] [PROTOCOL] [FAMILY] [IPTABLES] [LOCKINGOPT]'
-is_format_valid 'chain' 'port_ext' 'protocol'
+check_args '1' "$#" 'CHAIN [PORT] [PROTOCOL] [IPFAMILY] [IPTABLES] [FW_LOCKINGOPT]'
+is_format_valid 'chain' 'port_ext' 'protocol' 'ipfamily' 'iptables' 'fw_lockingopt'
```

Comment:
Do we really need iptables  / fw_lockingopt option

I don't we will ever want to implement anything other then iptables and it should all be the same..

Reply:
Maybe for iptables6 and iptables...

Reply:
this locking option (as -w) comes from standard example of fail2ban. Therefore I pass-through it here, like it is defined in standard library of fail2ban. Unfortunatelly it works actually only in case of "add chain". I do not know, why, but that were results of my "printf test" or better to say "echo test" and play with these scripts.
The problem with this additionally option is followed: If you ignore it completelly, may be, that it will be added during call as last argument. In this case you will have iptables="iptables -w" (or something else). Therefore we need a "very last" argument after "last argument" iptables. Then you will have "clean" separation in iptables="iptables" and rest in this "lockingopt".
After it we can decide, if we want to use it, if we call iptables. Actually it is only during "add chain" and I will observe, how it works. I do not know really, why fail2ban use this option. I will read more about it

Reply:
You can find this file under /etc/fail2ban/actions.d/iptables.conf
And that is a kind of "library" for calls over this iptables-interface

Reply:
`# Option:  lockingopt
# Notes.:  Option was introduced to iptables to prevent multiple instances from
#          running concurrently and causing irratic behavior.  -w was introduced
#          in iptables 1.4.20, so might be absent on older systems
#          See https://github.com/fail2ban/fail2ban/issues/1122
# Values:  STRING
lockingopt = -w

# Option:  iptables
# Notes.:  Actual command to be executed, including common to all calls options
# Values:  STRING
iptables = iptables <lockingopt>


[Init?family=inet6]

# Option:  blocktype (ipv6)
# Note:    This is what the action does with rules. This can be any jump target
#          as per the iptables man page (section 8). Common values are DROP
#          REJECT, REJECT --reject-with icmp6-port-unreachable
# Values:  STRING
blocktype = REJECT --reject-with icmp6-port-unreachable

# Option:  iptables (ipv6)
# Notes.:  Actual command to be executed, including common to all calls options
# Values:  STRING
iptables = ip6tables <lockingopt>`